platform-api-client-gwt/che-core-client-gwt-ssh/src/main/java/org/eclipse/che/api/ssh/gwt/client/SshServiceClientImpl.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2015 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.ssh.gwt.client;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+import org.eclipse.che.api.ssh.shared.dto.GenerateSshPairRequest;
+import org.eclipse.che.api.ssh.shared.dto.SshPairDTO;
+import org.eclipse.che.ide.dto.DtoFactory;
+import org.eclipse.che.ide.rest.AsyncRequestCallback;
+import org.eclipse.che.ide.rest.AsyncRequestFactory;
+import org.eclipse.che.ide.rest.RestContext;
+
+import javax.validation.constraints.NotNull;
+import java.util.List;
+
+/**
+ * The implementation of {@link SshServiceClient}.
+ *
+ * @author Sergii Leschenko
+ */
+public class SshServiceClientImpl implements SshServiceClient {
+    private final DtoFactory          dtoFactory;
+    private final AsyncRequestFactory asyncRequestFactory;
+    private final String              sshApi;
+
+    @Inject
+    protected SshServiceClientImpl(@RestContext String baseUrl,
+                                   DtoFactory dtoFactory,
+                                   AsyncRequestFactory asyncRequestFactory) {
+        this.dtoFactory = dtoFactory;
+        this.asyncRequestFactory = asyncRequestFactory;
+        sshApi = baseUrl + "/ssh";
+    }
+
+    @Override
+    public void getPairs(String service, @NotNull AsyncRequestCallback<List<SshPairDTO>> callback) {
+        asyncRequestFactory.createGetRequest(sshApi + "/" + service)
+                           .send(callback);
```

Comment:
I moved this class from che-plugins project. Thanks for remark I'll rework this class with promises